custom_components/versatile_thermostat/keep_alive.py

```diff
@@ -24,11 +24,16 @@ class IntervalCaller:
     Convenience wrapper around Home Assistant's `async_track_time_interval` function.
     """
 
-    def __init__(self, hass: HomeAssistant, interval_sec: int) -> None:
+    def __init__(self, hass: HomeAssistant, interval_sec: float) -> None:
```

Comment:
This change from `int` to `float` is not a fix for anything (I did not find anything to be broken). I simply realised that the `IntervalCaller` class would be more readily reusable for any purpose besides the keep-alive feature if it declared the interval member as a float, given that `async_track_time_interval` and `timedelta` are happy with fractional seconds.

I have intentionally _not_ changed the voluptuous schema, and the config flow UI still insists that the user should enter a positive integer for the keep-alive interval. That’s fine: I think this restriction makes sense for the feature “as seen by the user,” it’s just the `IntervalCaller` class that could be a bit more generic by handling floats.